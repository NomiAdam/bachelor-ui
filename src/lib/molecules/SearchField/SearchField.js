import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { IoClose } from 'react-icons/lib/io';
import { o, not } from 'ramda';
import { isNilOrEmptyString } from 'ramda-extension';
import Input from '../../atoms/Input/index';
import { lightTheme } from '../../constants/theme';

const SearchFieldWrapper = styled.div`
    background-color: ${ ({ backgroundColor }) => backgroundColor || lightTheme.lightGrey };
    border-color: transparent;
    border-radius: 50px;
    padding: 5px 15px;
    position: relative;
`;

const StyledIcon = styled(IoClose)`
    position: absolute;
    right: 10px;
    top: 7.5px;
    color: ${ lightTheme.lightBlue }
`;

const isNotNilOrEmptyString = o(not, isNilOrEmptyString);

const SearchField = ({ handleClear, value, ...others }) => (
    <SearchFieldWrapper>
        <Input
            { ...others }
            value={ value }
            borderBackground="transparent"
        />
        { isNotNilOrEmptyString(value) && <StyledIcon onClick={ handleClear } /> }
    </SearchFieldWrapper>
);

SearchField.propTypes = {
    /**
     * onChange function handler
     */
    onChange: PropTypes.func,
    /**
     * String value of TextField
     */
    value: PropTypes.any,
    /**
     * Whether Input is disabled or not
     */
    disabled: PropTypes.bool,
    /**
     * Function to be called upon onBlur event on input field
     */
    onBlur: PropTypes.func,
    /**
     * String definition of placeholder text
     */
    placeholder: PropTypes.string,
    /**
     * Function handle to clear whole text
     */
    handleClear: PropTypes.func,
    /**
     * String definition of background color
     */
    backgroundColor: PropTypes.string,
};

export { SearchFieldWrapper };
export default SearchField;
