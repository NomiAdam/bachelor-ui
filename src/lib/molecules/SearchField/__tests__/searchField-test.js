/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import SearchField, { SearchFieldWrapper } from '../SearchField';

describe('SearchField', () => {
    it('should be defined', () => {
        expect(SearchField).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<SearchField />)).toMatchSnapshot();
    });
    it('should be of type styled div', () => {
        expect(shallow(<SearchField/>).type()).toBe(SearchFieldWrapper);
    });
});
