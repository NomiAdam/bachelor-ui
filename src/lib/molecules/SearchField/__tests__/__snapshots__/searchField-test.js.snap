// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should match the snapshot 1`] = `
<styled.div>
  <Input
    borderBackground="transparent"
    type="text"
  />
</styled.div>
`;
